Detectors/TRD/workflow/src/TRDTrapSimulatorSpec.cxx

```diff
@@ -244,9 +242,19 @@ void TRDDPLTrapSimulatorTask::run(o2::framework::ProcessingContext& pc)
           mTrapSimulator[iTrap].filter();
           mTrapSimulator[iTrap].tracklet();
           trapSimAccumulatedTime += std::chrono::high_resolution_clock::now() - timeTrapProcessingStart;
-          nTrackletsInTrigRec += mTrapSimulator[iTrap].getTrackletArray64().size();
-          trapTrackletsAccum.insert(trapTrackletsAccum.end(), mTrapSimulator[iTrap].getTrackletArray64().begin(), mTrapSimulator[iTrap].getTrackletArray64().end());
-          // TODO get and output MC labels
+          auto trackletsOut = mTrapSimulator[iTrap].getTrackletArray64();
+          int nTrackletsOut = trackletsOut.size();
+          nTrackletsInTrigRec += nTrackletsOut;
+          auto digitCountOut = mTrapSimulator[iTrap].getTrackletDigitCount();
+          auto digitIndicesOut = mTrapSimulator[iTrap].getTrackletDigitIndices();
+          std::vector<o2::MCCompLabel> trackletLabels;
+          int currDigitIndex = 0;
+          for (int iTrklt = 0; iTrklt < nTrackletsOut; ++iTrklt) {
+            for (int iDigitIndex = currDigitIndex; iDigitIndex < digitCountOut[iTrklt]; ++iDigitIndex) {
+              trackletMCLabels.addElements(trapTrackletsAccum.size() + iTrklt, digitMCLabels.getLabels(digitIndicesOut[iDigitIndex]));
```

Comment:
Here one could look for duplicates. Not sure if we want it though. We could check the most frequent labels and state that this tracklet belonged primarily to the associated particle of those labels

Reply:
It would however make sense to have everything arrive at tracking and not have to be preprocessed, at least as far as possible.

Reply:
You mean in order to avoid preprocessing in the tracking we should remove the duplicates here? I think that is probably the best way, not sure if it would give any advantage to have some labels multiple times for the same tracklets

Reply:
I mean if its going to have to be done before tracking or the labels will need to be massaged in some way before tracking, it would seem more logical to do it here.

Reply:
Ok it's done. Not pretty, but I couldn't think of an easier way. There is a double loop comparing each new label with each label already added to the tracklet and only new labels are added. I assume for each digit the associated labels are unique, since also there duplicates are filtered in the digitization. Is that correct @jolopezl ?